python/lsst/faro/base/CatalogMeasurementBase.py

```diff
@@ -192,3 +196,27 @@ def _getReferenceCatalog(self, butlerQC, dataIds, refCats, filterList, epoch=Non
         refCatFrame = hstack([refCatTable, refCat.asAstropy()]).to_pandas()
 
         return refCatFrame
+
+    def _persistMeasurementInputs(self, config, shelveName, **kwargs):
+        """Persist in-memory objects sent as inputs to metric measurement run method.
+
+        This function is intended to be used for development and testing purposes
+        as a debug tool and is NOT to be used in routine operation.
+
+        Parameters
+        ----------
+        config : `lsst.pex.config.Config`
+            Config to be saved.
+        shelveName : `str`
+            Filename for output shelve.
+        """
+
+        import shelve
+
+        with shelve.open(shelveName, 'n') as shelf:
+            shelf['config'] = config
+            for key in kwargs.keys():
+                try:
+                    shelf[key] = kwargs[key]
+                except TypeError:
+                    print('ERROR shelving: {0}'.format(key))
```

Comment:
Why `print` and not `log` here?  We try to avoid `print` throughout, even for debugging code.

Also, is `TypeError` the correct exception?  I don't see any documented exceptions in the `shelve` docs, and the underlying `pickle` module returns its own errors.  Including any information from the exception, of whatever type, might also be useful.

Reply:
That's a good point. Removing this step of catching the exception.